### DISCUSSION TITLE
I want to connect with metamask through metamask connector in my mobile web browser.

### DISCUSSION THREAD

**Discussion by siosio34** [wevm/wagmi#2151] [Q&A] [answered]
I want to implement the following behavior using wagmi.

- If metamask is not installed on the mobile web, go to the metamask installation link
- If metamask is connected, when you call the connect function, if you press OK in metamask in the app, you will return to the original web page.

example service - https://looksrare.org/

wagmi is the cleanest to use among the libraries I've used.
However, the connection to Metamask does not work well in the mobile web browser.
How can I achieve the above behavior in a mobile web browser using wagmi?

**Comment by tmm** [comment, marked as answer]
Connectors in wagmi do not map 1:1 to wallets. The `MetaMaskConnector` in wagmi only works for MetaMask injected providers (e.g. browser extension, in-app browser). If you need to linking support on mobile, you want to use the `WalletConnectConnector`. In your app code, you could have some condition that detects a mobile device and uses the `WalletConnectConnector`, otherwise uses it uses the `MetaMaskConnector`.
